docs/development/core/server/kibana-plugin-server.coresetup.http.md

```diff
@@ -8,9 +8,9 @@
 
 ```typescript
 http: {
+        registerOnPreAuth: HttpServiceSetup['registerOnPreAuth'];
         registerAuth: HttpServiceSetup['registerAuth'];
-        registerOnRequest: HttpServiceSetup['registerOnRequest'];
-        getBasePathFor: HttpServiceSetup['getBasePathFor'];
-        setBasePathFor: HttpServiceSetup['setBasePathFor'];
+        registerOnPostAuth: HttpServiceSetup['registerOnPostAuth'];
+        basePath: HttpServiceSetup['basePath'];

```

Comment:
@legrego I've changed api slightly to keep the same pattern where we group functionality by domain term namespace.